### PR TITLE
add build.sh to allow 1-liner to build development version of the compiler; can be rerun safely thanks to [ -d csources ]

### PR DESCRIPTION
**Pull request by timotheecour**

### CODE REVIEW

**Comment by Araq** [comment]
But the old instructions are clearer; you hide the fact the external ``csources`` project is used. If these scripts do too magic, it's not clear to outsiders how to fix upcoming problems.

Maybe there is a way to get the best of both worlds?

File: readme.md [hunk 1]
```diff
@@ -50,9 +50,8 @@ Next, to build from source you will need:
     other distros as well). 
 
 Then, if you are on a \*nix system or Windows, the following steps should compile
-Nim from source using ``gcc``, ``git`` and the ``koch`` build tool (in the place
-of ``sh build.sh`` you should substitute ``build.bat`` on x86 Windows or
-``build64.bat`` on x86_64 Windows):
+Nim from source using ``gcc``, ``git`` and the ``koch`` build tool (for Windows,
+see adjustments needed in `build_all.sh`).
```
Comment (Araq) [inline readme.md]: I think Windows users don't enjoy looking into shell scripts and feel like second class citizens.

Comment (timotheecour) [reply readme.md]: done: see latest commit

**Comment by timotheecour** [comment]
PTAL

File: build_all.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#! /bin/sh
+
+# build development version of the compiler; can be rerun safely
+
+set -u
```
Comment (Araq) [inline build_all.sh]: Add a comment what this command does. And ensure it's in Posix.

File: build_all.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+set -e
```
Comment (Araq) [inline build_all.sh]: Same.

File: build_all.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+echo_run(){
```
Comment (Araq) [inline build_all.sh]: Do Posix shell scripts support functions?

Comment (timotheecour) [reply build_all.sh]: yes

File: build_all.sh (hunk 3 above)
Comment (timotheecour) [reply build_all.sh]: done

File: build_all.sh (hunk 2 above)
Comment (timotheecour) [reply build_all.sh]: done; set -e, set -u are standard posix stuff

**Comment by timotheecour** [comment]
PTAL, all comments addressed

**Pull request merged**